packages/grid/_modules_/grid/components/menu/columnMenu/HideGridColMenuItem.tsx

```diff
@@ -4,15 +4,16 @@ import MenuItem from '@mui/material/MenuItem';
 import { GridFilterItemProps } from './GridFilterItemProps';
 import { useGridApiContext } from '../../../hooks/utils/useGridApiContext';
 import { useGridRootProps } from '../../../hooks/utils/useGridRootProps';
-import { visibleGridColumnsLengthSelector } from '../../../hooks/features/columns';
+import { visibleGridColumnsSelector } from '../../../hooks/features/columns';
 
 const HideGridColMenuItem = (props: GridFilterItemProps) => {
   const { column, onClick } = props;
   const apiRef = useGridApiContext();
   const rootProps = useGridRootProps();
   const timeoutRef = React.useRef<any>();
 
-  const visibleColumnsLength = visibleGridColumnsLengthSelector(apiRef.current.state);
+  const visibleColumns = visibleGridColumnsSelector(apiRef.current.state);
+  const columnsWithMenu = visibleColumns.filter((col) => col.disableColumnMenu !== true);
```

Comment:
Can we create a dedicated `gridVisibleColumnsWithColumnMenu` selector, that way we only loop through the columns once per render, not once per column per render.